projects/dotcms-field-elements/src/components/dot-form/dot-form.tsx

```diff
@@ -0,0 +1,91 @@
+import { Component, Element, Event, EventEmitter, Listen, Prop } from '@stencil/core';
+import { DotCMSContentTypeField } from '../../../../dotcms/src/lib/models';
+import { DotFormFields } from './dot-form-fields';
+
+const fieldMap = {
+    Text : ((fieldParam: DotCMSContentTypeField) => DotFormFields.Text(fieldParam)),
+    Checkbox : ((fieldParam: DotCMSContentTypeField) => DotFormFields.Checkbox(fieldParam)),
+    Select : ((fieldParam: DotCMSContentTypeField) => DotFormFields.Select(fieldParam))
+};
+
+@Component({
+    tag: 'dot-form'
+})
+export class DotFormComponent {
+    @Element() el: HTMLElement;
+    @Event() formSubmit: EventEmitter;
+    @Prop() fields = [];
+    @Prop() fieldsToShow = [];
+    @Prop() resetLabel = 'Reset';
+    @Prop() submitLabel = 'Submit';
+
+    @Prop({ mutable: true }) _formValues = {};
+
+    @Listen('valueChanges')
+    onValueChanges(event: any): void {
+        this._formValues[event.detail.name] = event.detail.value;
+    }
+
+    @Listen('stateChanges')
+    onStateChanges(event: CustomEvent): void {
+        // refresh variables from hostData
+    }
+
+    hostData() {
+        // TODO: do validation here
+        return {
+          'class': { 'is-open': this._formValues },
+          'aria-hidden': this._formValues ? 'false' : 'true'
+        };
+      }
+
+    handleSubmit(evt: Event): void {
```

Comment:
doc in all the methods.

Reply:
all those methods are private, the only public are the ones decorated with `@Method()`